packages/vscode-dependencies-validation/src/npmIssuesActionProvider.ts

```diff
@@ -0,0 +1,54 @@
+import type {
+  CancellationToken,
+  CodeActionContext,
+  CodeActionProvider,
+  Command,
+  Diagnostic,
+  Range,
+  Selection,
+  TextDocument,
+  CodeAction,
+  CodeActionKind,
+} from "vscode";
+import { get } from "lodash";
+import {
+  NPM_DEPENDENCY_ISSUES_CODE,
+  FIX_ALL_ISSUES_COMMAND,
+} from "./constants";
+
+export class NPMIssuesActionProvider implements CodeActionProvider {
+  constructor(private kind: CodeActionKind) {}
+
+  public provideCodeActions(
+    document: TextDocument,
+    range: Range | Selection,
+    context: CodeActionContext,
+    token: CancellationToken
+  ): CodeAction[] {
+    // for each diagnostic entry create a code action command
+    return context.diagnostics
+      .filter((diagnostic) => diagnostic.code === NPM_DEPENDENCY_ISSUES_CODE)
+      .map((diagnostic) => this.createCommandCodeAction(diagnostic));
+  }
+
+  private createCommandCodeAction(diagnostic: Diagnostic): CodeAction {
+    const action: CodeAction = {
+      title: "Fix all dependency issues",
+      kind: this.kind,
+    };
+
+    action.command = createCommand(diagnostic);
+    action.diagnostics = [diagnostic];
+    action.isPreferred = true;
+
+    return action;
+  }
+}
+
+function createCommand(diagnostic: Diagnostic): Command {
+  return {
+    command: FIX_ALL_ISSUES_COMMAND,
+    title: "Fix all dependency issues",
+    arguments: [get(diagnostic, "packageJsonPath")],
```

Comment:
we should check in UI5-Lang-assistanst if/how we linked commands and diagnostics.
Although that also had LSP in the middle.